lib/std/zig/CrossTarget.zig

```diff
@@ -354,20 +354,20 @@ pub fn parseCpuArch(args: ParseOptions) ?Target.Cpu.Arch {
     }
 }
 
-/// Parses a version with an omitted patch component, such as "1.0",
-/// which SemanticVersion.parse is not capable of.
-fn parseVersion(ver: []const u8) !SemanticVersion {
+/// Similar to `SemanticVersion.parse`, but also handles cases:
+/// * When `patch` component is omitted, such as `1.0` or `2.37`. In this case `patch` will be 0.
+/// * Leading zeroes are allowed.
+pub fn parseVersion(ver: []const u8) error{ InvalidVersion, Overflow }!SemanticVersion {
     const parseVersionComponent = struct {
```

Comment:
IDK what do you precisely mean here, but I renamed it to "parseVersionComponentFn" and put `struct { ... }` in parentheses.

Reply:
I'm probably just confused.  I don't understand why there is a `struct` declaration here at all.  I am guessing its a hack to effectively declare a nested function inside parseVersion.  But I'm curious if its accomplishing something else.

(Seems cleaner to just define a top-level, non-public function to me, but I'm not clear if there is some Zig style or other motivation for the current approach.)

Anyway, probably not a discussion to have on a PR.  I should just bring this up in Discord or somesuch.

The fixes and test cases look great, btw.

Reply:
> I'm probably just confused.  I don't understand why there is a `struct` declaration here at all.  I am guessing its a hack to effectively declare a nested function inside parseVersion.  But I'm curious if its accomplishing something else.
> 
> (Seems cleaner to just define a top-level, non-public function to me, but I'm not clear if there is some Zig style or other motivation for the current approach.)
> 
> Anyway, probably not a discussion to have on a PR.  I should just bring this up in Discord or somesuch.
> 
> The fixes and test cases look great, btw.

Oh, sorry I misunderstood it. IIRC the reason was to not pollute this upper namespace with function that is used in only one function and allowing to use same identifier in other functions (with top-level function you would've need to rename new declarations to avoid shadowing). You can define sub-namespace `std.CrossTarget.Version` tho:

```zig
// maybe different name, idk.
pub const Version = struct {
    /// current docs
    pub fn parse(ver: []const u8) error{ InvalidVersion, Overflow }!SemanticVersion {
        // current parseVersion implementation
    }

    fn parseComponent(component: []const u8) error{ InvalidVersion, Overflow }!usize {
        // current parseVersionComponentInner implementation
        // This way parseComponent does not pollute upper namespace
    }
};

// To avoid unneccessary breakage, with deprecation comment
pub const parseVersion = Version.parse;
// std.zig.CrossTarget.versionParse becomes std.zig.CrossTarget.Version.parse
```

But IMHO this goes out-of-scope of this PR and would require more time for reviewings.

Thank you for your reviews, they were very helpful :)